eox_tenant/models.py

```diff
@@ -65,6 +66,59 @@ def get_microsite_for_domain(cls, domain):
         return microsites[0] if microsites else None
 
 
+class TenantConfig(models.Model):
+    """
+    Model to persist edxapp configurations.
+    """
+    key = models.CharField(max_length=63, db_index=True)
+    values = JSONField(null=False, blank=True, load_kwargs={'object_pairs_hook': collections.OrderedDict})
+    lms_configs = JSONField(null=False, blank=True, load_kwargs={'object_pairs_hook': collections.OrderedDict})
+    studio_configs = JSONField(null=False, blank=True, load_kwargs={'object_pairs_hook': collections.OrderedDict})
+    theming_configs = JSONField(null=False, blank=True, load_kwargs={'object_pairs_hook': collections.OrderedDict})
+    sites = models.ManyToManyField(Site)
+
+    class Meta:
+        """
+        Model meta class.
+        """
+        app_label = "eox_tenant"
+
+    def __unicode__(self):
+        return self.key
+
+    def get_organizations(self):
+        """
+        Helper method to get organizations.
+        """
+
+        org_filter = self.values.get('course_org_filter')  # pylint: disable=no-member
+
+        if isinstance(org_filter, str):
+            org_filter = [org_filter]
+
+        return org_filter
+
+    @classmethod
+    def get_configs_for_domain(cls, domain):
+        """
+        Get edxapp configuration using a domain. There is a compat layer to support microsite until
+        deprecation.
+        """
+
+        # remove any port number from the hostname
+        domain = domain.split(':')[0]
+        config = TenantConfig.objects.filter(sites__domain=domain).last()  # pylint: disable=no-member
+
+        if config:
+            return config.values, config.key
+
+        microsite = Microsite.get_microsite_for_domain(domain)
```

Comment:
it is weird to have this backwards compatibility here. I would rather put it on the `get_tenant_config_by_domain` method of the backends.database module